local/experimental/blockchain.go

```diff
@@ -0,0 +1,490 @@
+				return err
+			}
+			if err := ln.HealthyUnsafe(ctx); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (w *wallet) createBlockchains(
+	ctx context.Context,
+	chainSpecs []network.BlockchainSpec,
+	blockchainTxs []*txs.Tx,
+) error {
+	w.log.Info(logging.Green.Wrap("creating each custom chain"))
+	for i, chainSpec := range chainSpecs {
+		vmName := chainSpec.VMName
+		vmID, err := utils.VMID(vmName)
+		if err != nil {
+			return err
+		}
+		w.log.Info("creating blockchain",
+			zap.String("vm-name", vmName),
+			zap.String("vm-ID", vmID.String()),
+		)
+
+		cctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+		defer cancel()
+
+		blockchainID, err := w.wallet.IssueTx(
+			blockchainTxs[i],
+			common.WithContext(cctx),
+			defaultPoll,
+		)
+		if err != nil {
+			return fmt.Errorf("failure issuing create blockchain: %w", err)
+		}
+		if blockchainID != blockchainTxs[i].ID() {
+			return fmt.Errorf("failure issuing create blockchain: txID differs from blockchainID")
+		}
+
+		w.log.Info("created a new blockchain",
+			zap.String("vm-name", vmName),
+			zap.String("vm-ID", vmID.String()),
+			zap.String("blockchain-ID", blockchainID.String()),
+		)
+	}
+	return nil
+}
+
+func (w *wallet) addSubnetValidators(
+	ctx context.Context,
+	ln LocalNetwork,
+	chainSpecs []network.BlockchainSpec,
+) error {
+	w.log.Info(logging.Green.Wrap("adding the nodes as subnet validators"))
+	cctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	vs, err := w.client.GetCurrentValidators(cctx, constants.PrimaryNetworkID, nil)
+	cancel()
+	if err != nil {
+		return err
+	}
+	primaryValidatorsEndtime := make(map[ids.NodeID]time.Time)
+	for _, v := range vs {
+		primaryValidatorsEndtime[v.NodeID] = time.Unix(int64(v.EndTime), 0)
+	}
+	for _, spec := range chainSpecs {
+		subnetID, err := ids.FromString(*spec.SubnetID)
+		if err != nil {
+			return err
+		}
+		nodeIDs := make([]ids.NodeID, len(spec.Participants))
+		for i, nodeName := range spec.Participants {
+			node := ln.GetNodeUnsafe(nodeName)
+			nodeID := node.GetNodeID()
+			nodeIDs[i] = nodeID
+			cctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+			txID, err := w.wallet.IssueAddSubnetValidatorTx(
+				&txs.SubnetValidator{
+					Validator: txs.Validator{
+						NodeID: nodeID,
+						// reasonable delay in most/slow test environments
+						Start: uint64(time.Now().Add(validationStartOffset).Unix()),
+						End:   uint64(primaryValidatorsEndtime[nodeID].Unix()),
+						Wght:  subnetValidatorsWeight,
+					},
+					Subnet: subnetID,
+				},
+				common.WithContext(cctx),
+				defaultPoll,
+			)
+			cancel()
+			if err != nil {
+				return err
+			}
+			w.log.Info("added node as a subnet validator",
+				zap.String("name", nodeName),
+				zap.String("nodeID", nodeID.String()),
+				zap.String("subnetID", subnetID.String()),
+				zap.String("txID", txID.String()),
+			)
+		}
+
+		// Wait for validators to become active
```

Comment:
also maybe this can be split, as a network may want to do something in the middle